homeassistant/components/rachio/switch.py

```diff
@@ -219,6 +242,16 @@ async def async_added_to_hass(self):
                 KEY_RAIN_DELAY
             ] / 1000 > as_timestamp(now())
 
+        # If the controller was in a rain delay state during a reboot, this re-sets the timer
+        if self._state is True:
+            delay_end = utc_from_timestamp(
+                self._controller.init_data[KEY_RAIN_DELAY] / 1000
+            )
+            _LOGGER.debug("Re-setting rain delay timer for %s", delay_end)
+            self.cancel_update = async_track_point_in_utc_time(
```

Comment:
Is it possible a timer is already running here that needs to be canceled before a new one is created?

Reply:
In testing the timer didn't persist after a reboot, which is why its called again here.

Reply:
Right, it shouldn't persist between reboots.

For clarity, I'm asking if it is possible that `_async_handle_update` could have set the timer before `turn_on` is called?

Reply:
I might be confused (almost certainly), but at this point `self._state` is being set by 
```
if KEY_RAIN_DELAY in self._controller.init_data:
            self._state = self._controller.init_data[
                KEY_RAIN_DELAY
            ] / 1000 > as_timestamp(now())
```
which would happen before any updates happened. Or am I missing something?

Reply:
I had a moment to read over it again without any distractions.  It looks good to go. 👍